libnavigation-core/src/test/java/com/mapbox/navigation/core/reroute/MapboxRerouteControllerTest.kt

```diff
@@ -0,0 +1,290 @@
+package com.mapbox.navigation.core.reroute
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.routeoptions.RouteOptionsProvider
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.testing.MainCoroutineRule
+import com.mapbox.navigation.utils.internal.ThreadController
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import io.mockk.verifyOrder
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class MapboxRerouteControllerTest {
+
+    private lateinit var rerouteController: MapboxRerouteController
+
+    @MockK
+    private lateinit var directionsSession: DirectionsSession
+
+    @MockK
+    private lateinit var tripSession: TripSession
+
+    @MockK
+    private lateinit var routeOptionsProvider: RouteOptionsProvider
+
+    @MockK
+    private lateinit var logger: Logger
+
+    // sealed class RouteOptionsProvider.RouteOptionsResult *******/
```

Comment:
NIT Do we need these 👇 comments?

Reply:
why not? I could  see by 👀 what mock is it.

Reply:
Isn't that clear enough? I mean why do we need to note that it's for `RouteOptionsProvider.RouteOptionsResult`?

Reply:
fixed

Reply:
Did you push? Comments are still present 🤔

Reply:
fixed